mmeval/metrics/bleu.py

```diff
@@ -0,0 +1,193 @@
+        {'bleu': ...}
+    """
+
+    def __init__(self,
+                 n_gram: int = 4,
+                 smooth: bool = False,
+                 ngram_weights: Optional[Sequence[float]] = None,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.n_gram = n_gram
+        self.smooth = smooth
+        if ngram_weights is not None and len(ngram_weights) != n_gram:
+            raise ValueError(
+                f'List of weights has different weights than `n_gram`: '
+                f'{len(ngram_weights)} != {n_gram}')
+        if ngram_weights is None:
+            ngram_weights = [1.0 / n_gram] * n_gram
+        self.ngram_weights = ngram_weights
+
+    def add(self, predictions: Sequence[str], references: Sequence[Sequence[str]]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add the intermediate results to ``self._results``.
+
+        Args:
+             predictions (Sequence[str]): An iterable of machine
+                translated corpus.
+             references (Sequence[Sequence[str]]): An iterable of
+                iterables of reference corpus.
+        """
+
+        references_token: Sequence[Sequence[Sequence[str]]] = [[
+            tokenizer_fn(line) if line is not None else [] for line in r
+        ] for r in references]
+        predictions_token: Sequence[Sequence[str]] = [
+            tokenizer_fn(line) if line else [] for line in predictions
+        ]
+        pred_len = 0
+        references_len = 0
+        for prediction, references in zip(predictions_token, references_token):
+            pred_len = len(prediction)
+            references_len_list = [len(reference) for reference in references]
+            references_len_diff = [
+                abs(len(prediction) - length) for length in references_len_list
+            ]
+            # In the multi sentence reference, the one whose length is closest
+            # to the predicted sentence is selected to record the length.
+            min_index = references_len_diff.index(min(references_len_diff))
+            references_len = references_len_list[min_index]
+
+            pred_counter: Counter = get_n_gram(prediction, self.n_gram)
+            reference_counter: Counter = Counter()
+            for reference in references:
+                # Take intersection for the n_gram of references.
+                reference_counter |= get_n_gram(reference, self.n_gram)
+            # Union the n_gram of prediction and references.
+            counter_clip = pred_counter & reference_counter
+            precision_matches = np.zeros(self.n_gram)
+            precision_total = np.zeros(self.n_gram)
+            for counter in counter_clip:
+                precision_matches[len(counter) - 1] += counter_clip[counter]
+            for counter in pred_counter:
+                precision_total[len(counter) - 1] += pred_counter[counter]
+            result = (pred_len, references_len, precision_matches,
```

Comment:
Suggesting add a blank line here~
```suggestion
                precision_total[len(counter) - 1] += pred_counter[counter]

            result = (pred_len, references_len, precision_matches,
```